llvm/test/CodeGen/AMDGPU/sink-image-sample.ll

```diff
@@ -11,7 +11,7 @@
 
 define amdgpu_ps float @sinking_img_sample() {
 main_body:
-  %i = call <3 x float> @llvm.amdgcn.image.sample.2d.v3f32.f32(i32 7, float undef, float undef, <8 x i32> undef, <4 x i32> undef, i1 false, i32 0, i32 0)
+  %i = call <3 x float> @llvm.amdgcn.image.sample.l.2d.v3f32.f32(i32 7, float undef, float undef, float undef, <8 x i32> undef, <4 x i32> undef, i1 false, i32 0, i32 0)
```

Comment:
Add a new test instead of changing the operation in an old one? Presumably we should have tests for all the cases

Reply:
I added more tests to cover other cases of image.sample